Packs/XMatters/Integrations/xMatters/xMatters.py

```diff
@@ -0,0 +1,801 @@
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: Username and Password fields and verify the user exists'
+        else:
+            raise e
+
+    return 'ok'
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    instance = demisto.params().get('instance')
+    username = demisto.params().get('username')
+    password = demisto.params().get('password')
+    property_name = demisto.params().get('property_name')
+    property_value = demisto.params().get('property_value')
+    base_url = demisto.params().get('url')
+
+    # if your Client class inherits from BaseClient, SSL verification is
+    # handled out of the box by it, just pass ``verify_certificate`` to
+    # the Client constructor
+    verify_certificate = not demisto.params().get('insecure', False)
+
+    # How much time before the first fetch to retrieve incidents
+    first_fetch_time = arg_to_timestamp(
+        arg=demisto.params().get('first_fetch', '3 days'),
+        arg_name='First fetch time',
+        required=True
+    )
+    # Using assert as a type guard (since first_fetch_time is always an int when required=True)
+    assert isinstance(first_fetch_time, int)
+
+    # if your Client class inherits from BaseClient, system proxy is handled
+    # out of the box by it, just pass ``proxy`` to the Client constructor
+    proxy = demisto.params().get('proxy', False)
+
+    # INTEGRATION DEVELOPER TIP
+    # You can use functions such as ``demisto.debug()``, ``demisto.info()``,
+    # etc. to print information in the XSOAR server log. You can set the log
+    # level on the server configuration
+    # See: https://xsoar.pan.dev/docs/integrations/code-conventions#logging
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+
+        to_xm_client = Client(
+            base_url=base_url,
+            verify=verify_certificate,
+            auth=(username, password),
+            proxy=proxy)
+
+        from_xm_client = Client(
+            base_url="https://" + instance,
+            verify=verify_certificate,
+            auth=(username, password),
+            proxy=proxy)
+
+        if demisto.command() == 'xm-trigger-workflow':
+            return_results(xm_trigger_workflow_command(
+                to_xm_client,
+                demisto.args().get('recipients'),
+                demisto.args().get('subject'),
+                demisto.args().get('body'),
+                demisto.args().get('incident_id'),
+                demisto.args().get('close_task_id')
+            ))
+        elif demisto.command() == 'fetch-incidents':
+            # Set and define the fetch incidents command to run after activated via integration settings.
+            alert_status = demisto.params().get('status', None)
+            priority = demisto.params().get('priority', None)
+            max_fetch = demisto.params().get('max_fetch', 20)
```

Comment:
We reccomend enforcing a max limit of 200 incidents per fetch as to not overload XSOAR with too many incidents at once. For more information see [here](https://xsoar.pan.dev/docs/integrations/fetching-incidents#fetch-limit)